crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -60,97 +61,74 @@ impl std::cmp::PartialEq<&Visibility> for Visibility {
+    pub const HIDDEN: Self = Self(false);
+    /// An entity that is visible in the hierarchy.
+    pub const VISIBLE: Self = Self(true);
+
+    /// Returns `true` if the entity is visible in the hierarchy.
+    /// Otherwise, returns `false`.
+    pub fn get(self) -> bool {
+        self.0
     }
 }
 
-impl ComputedVisibility {
-    /// A [`ComputedVisibility`], set as invisible.
-    pub const HIDDEN: Self = ComputedVisibility {
-        flags: ComputedVisibilityFlags::empty(),
-    };
-
-    /// Whether this entity is visible to something this frame. This is true if and only if [`Self::is_visible_in_hierarchy`] and [`Self::is_visible_in_view`]
-    /// are true. This is the canonical method to call to determine if an entity should be drawn.
-    /// This value is updated in [`PostUpdate`] by the [`VisibilitySystems::CheckVisibility`] system set.
-    /// Reading it during [`Update`](bevy_app::Update) will yield the value from the previous frame.
-    #[inline]
-    pub fn is_visible(&self) -> bool {
-        self.flags.bits() == ComputedVisibilityFlags::all().bits()
-    }
-
-    /// Whether this entity is visible in the entity hierarchy, which is determined by the [`Visibility`] component.
-    /// This takes into account "visibility inheritance". If any of this entity's ancestors (see [`Parent`]) are hidden, this entity
-    /// will be hidden as well. This value is updated in the [`VisibilitySystems::VisibilityPropagate`], which lives in the [`PostUpdate`] schedule.
-    #[inline]
-    pub fn is_visible_in_hierarchy(&self) -> bool {
-        self.flags
-            .contains(ComputedVisibilityFlags::VISIBLE_IN_HIERARCHY)
-    }
-
-    /// Whether this entity is visible in _any_ view (Cameras, Lights, etc). Each entity type (and view type) should choose how to set this
-    /// value. For cameras and drawn entities, this will take into account [`RenderLayers`].
-    ///
-    /// This value is reset to `false` every frame in [`VisibilitySystems::VisibilityPropagate`] during [`PostUpdate`].
-    /// Each entity type then chooses how to set this field in the [`VisibilitySystems::CheckVisibility`] system set, in [`PostUpdate`].
-    /// Meshes might use frustum culling to decide if they are visible in a view.
-    /// Other entities might just set this to `true` every frame.
-    #[inline]
-    pub fn is_visible_in_view(&self) -> bool {
-        self.flags
-            .contains(ComputedVisibilityFlags::VISIBLE_IN_VIEW)
+/// Algorithmically-computed indication or whether an entity is visible and should be extracted for rendering.
+///
+/// This will be reset to `false` at the beginning of [`PostUpdate`], and then set to true by
+/// [`CheckVisibility`] systems. Because of this, values of this type will be marked as changed on
+/// nearly every frame, even when they do not change.
+///
+/// [`VisibilitySystems::CheckVisibility`]
+#[derive(Component, Deref, Debug, Default, Clone, Copy, Reflect, PartialEq, Eq)]
+pub struct ViewVisibility(bool);
```

Comment:
Given that we have a trivial getter, and a public way to set values as both true and false, I have a hard time seeing why we shouldn't just make this field pub.

The docs about how to use this are excellent, but they could just be moved to the struct level.

Reply:
In previous discussion about components like `Parent`, I think it was decided that we should avoid the `.0` syntax in public APIs, which is why I opted to derive deref and add a getter.

I decided not to provide a `.set(false)` option, as marking an entity as visible in view is meant to be irreversible during a given frame (this is a property I carried over from the old API, which had `.set_visible_in_view()`). Of course you could still reset the view visibility by just overwriting the entire component, but at least this API nudges users in the right direction.

Agreed about the documentation.